packages/host/app/components/realm-dropdown.gts

```diff
@@ -0,0 +1,71 @@
+import Component from '@glimmer/component';
+import { service } from '@ember/service';
+
+import { DropdownButton } from '@cardstack/boxel-ui/components';
+import { type RealmInfo, RealmPaths } from '@cardstack/runtime-common';
+
+import type RealmInfoService from '../services/realm-info-service';
+
+export interface RealmDropdownItem extends RealmInfo {
+  path: string;
+  iconURL: string | null;
+}
+
+interface Signature {
+  Args: {
+    onSelect: (item: RealmDropdownItem) => void;
+    selectedRealmURL: URL | undefined;
+    disabled?: boolean;
+  };
+  Element: HTMLElement;
+}
+
+export default class RealmDropdown extends Component<Signature> {
+  <template>
+    <DropdownButton
+      class='realm-dropdown-button'
+      @items={{this.realms}}
+      @onSelect={{@onSelect}}
+      @selectedItem={{this.selectedRealm}}
+      @disabled={{@disabled}}
+      @kind='secondary-light'
+      @size='small'
+      data-test-realm-dropdown
+      ...attributes
+    >
+      {{if this.selectedRealm 'Change' 'Select'}}
+    </DropdownButton>
+  </template>
+
+  @service declare realmInfoService: RealmInfoService;
+
+  constructor(owner: unknown, args: Signature['Args']) {
+    super(owner, args);
+    this.realmInfoService.fetchAllKnownRealmInfos.perform();
+  }
+
+  get realms(): RealmDropdownItem[] {
+    let items: RealmDropdownItem[] | [] = [];
+    for (let [
+      path,
+      realmInfo,
+    ] of this.realmInfoService.cachedRealmInfos.entries()) {
```

Comment:
I'm not sure what's the best course of action here. I make a call to fetch everything when the dropdown opens, but it might take a while for any missing realm to appear. I could put a loading icon until all realms are fetched... I considered making the realm info service fetch everything from its construction. However that might slow everything else down because this service is used a lot.